templates/test/by_id.html

```diff
@@ -135,7 +138,29 @@ <h4 class="category-header {{ 'fail' if result.error else 'pass' }}" data-catego
     {{ super() }}
     <script type="text/javascript" src="{{ url_for('static', filename='js/same-blocks-highlight.js') }}"></script>
     <script type="text/javascript">
+
         $(document).ready(function(){
+            {% if not test.finished %}
+            setInterval(function(){ 
+            $.ajax({
+            type:"GET",
+            url: "{{ url_for('.by_id', test_id=test.id) }}",
+            contentType:"application/json; charset=utf-8",
+            success:function (data) {
+                var result = $('<div />').append(data).find('#progress_details').html();
```

Comment:
Fetching the entire page and just replacing the html is wasting resources.

It's better to make a new route that returns some json with the test status, estimated time left and test progress entries. If the test is finished, reload the page, otherwise update the single components needed.